drift-transport-netty/src/main/java/com/facebook/drift/transport/netty/codec/ZlibTransformer.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (C) 2013 Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.drift.transport.netty.codec;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.zip.DataFormatException;
+import java.util.zip.Deflater;
+import java.util.zip.Inflater;
+
+public class ZlibTransformer
+        implements ByteBufTransformer
+{
+    private static final int ZLIB_COMPRESSION_BUFFER_SIZE = 512;
+    public static final ZlibTransformer ZLIB_TRANSFORMER = new ZlibTransformer();
+
+    @Override
+    public ByteBuf transform(ByteBuf data)
+    {
+        return compress(data);
+    }
+
+    @Override
+    public ByteBuf untransform(ByteBuf data)
+    {
+        return uncompress(data);
+    }
+
+    private static ByteBuf compress(ByteBuf data)
```

Comment:
I think this is a simpler version of this method. Would it work?

```java
    private static ByteBuf compress(ByteBuf data)
    {
        Deflater compressor = new Deflater();
        int inputLength = data.readableBytes();
        ByteArrayOutputStream baos = new ByteArrayOutputStream(inputLength + ZLIB_COMPRESSION_BUFFER_SIZE);
        try (DeflaterOutputStream deflaterOutputStream = new DeflaterOutputStream(baos, compressor)) {
            data.readBytes(deflaterOutputStream, data.readableBytes());
        }
        catch (IOException e) {
            throw new UncheckedIOException(e);
        }
        finally {
            compressor.end();
            data.release();
        }
        return Unpooled.wrappedBuffer(baos.toByteArray());
    }
```

Reply:
Hmm, interesting, let me try.